### PR TITLE
Add new formula: dovecot-pigeonhole

### PR DESCRIPTION
**Pull request by wahjava**
I created this formula for my personal use, so decided to contribute it. This is my first contribution to this project, so let me know if something needs change.

Thanks!

### CODE REVIEW

File: Library/Formula/dovecot-pigeonhole.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    if build.with? "unfinished-features"
+      args << "--with-unfinished-features"
+    end
+    system "./configure", *args
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    system "false"
```
Comment (bfontaine) [inline Library/Formula/dovecot-pigeonhole.rb]: Please add a test here.

File: Library/Formula/dovecot-pigeonhole.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  depends_on "dovecot"
+
+  def install
+    args = ["--disable-dependency-tracking",
+            "--with-dovecot=#{HOMEBREW_PREFIX}/lib/dovecot",
+            "--prefix=#{prefix}",]
+
+    if build.with? "unfinished-features"
+      args << "--with-unfinished-features"
+    end
```
Comment (bfontaine) [inline Library/Formula/dovecot-pigeonhole.rb]: You can inline this:
```rb
args << "--with-unfinished-features" if build.with? "unfinished-features"
```

File: Library/Formula/dovecot-pigeonhole.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  homepage "http://pigeonhole.dovecot.org/"
+  url "http://pigeonhole.dovecot.org/releases/2.2/dovecot-2.2-pigeonhole-0.4.8.tar.gz"
+  sha256 "d73c1c5a11cdfdcb58304a1c1272cce6c8e1868e3f61d393b3b8a725f3bf665b"
+
+  option "with-unfinished-features", "Build unfinished new features/extensions"
+
+  depends_on "dovecot"
+
+  def install
+    args = ["--disable-dependency-tracking",
```
Comment (DomT4) [inline Library/Formula/dovecot-pigeonhole.rb]: It's a nit, but this style looks a bit neater:

```ruby
args = %W[
  --disable-dependency-tracking
  --with-dovecot=#{HOMEBREW_PREFIX}/lib/dovecot
  --prefix=#{prefix}
]
```

**Comment by DomT4** [comment]
Ping?

**Comment by wahjava** [comment]
Apologies for going MIA. Thanks for the feedback on this. I'll incorporate that. But, as I look in the CI build output, it does not seem to comply with sandboxing mechanism:

    Sep 25 08:09:18 sandboxd[124]: mkdir(13960) deny file-write-create /usr/local/Cellar/dovecot/2.2.18/include/dovecot/sieve
    Sep 25 08:09:18 sandboxd[124]: mkdir(13963) deny file-write-create /usr/local/Cellar/dovecot/2.2.18/include/dovecot/sieve

I'm not really sure how to go about fixing this, nor I'm aware of any such formulae, where addons/extensions are provided by separate formula. Any ideas ?

Should I rather submit a diff for incorporating it in dovecot formula as an option ?

Thanks!

**Comment by mikemcquaid** [comment]
> I'm not really sure how to go about fixing this, nor I'm aware of any such formulae, where addons/extensions are provided by separate formula. Any ideas ?

In these types of cases (formula B needs to write into the cellar of formula A) it should instead be offered as an option on the `dovecot` formula and use a `resource` there.

**Comment by leafac** [comment]
I took a stab at the issue [here](https://github.com/Homebrew/homebrew/pull/46331). I'm using the modified formula and it's working for me.

**Pull request closed without merging**